agent/puppetupdate.rb

```diff
@@ -205,9 +216,15 @@ def resolve(git, env, limit=nil)
                 changes << msg
               end
             else
-              msg = reset_ref(ref, sha)
-              Log.info msg
-              changes << msg
+              time = run(git_cmd('show --quiet --format=format:%%at %s', ref)).to_i
+
+              if time < expiration_time
+                Log.info "ignoring #{dir}: older than #{expire_after_days} days"
```

Comment:
Don't worry, I _was_ reading it wrong. It's hard to follow the script until you've read most of it through :)